integration/read_write_mode_test.go

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+//go:build requires_docker
+// +build requires_docker
+
+package integration
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+)
+
+func TestReadWriteMode(t *testing.T) {
+	s, err := e2e.NewScenario(networkName)
+	require.NoError(t, err)
+	defer s.Close()
+
+	minio := e2edb.NewMinio(9000, blocksBucketName, alertsBucketName, rulerBucketName)
+	require.NoError(t, s.StartAndWaitReady(minio))
+
+	flags := map[string]string{
```

Comment:
I would rather use my `CommonStorageBackendFlags()`, and remove `-alertmanager-storage.s3.bucket-name` and `-ruler-storage.s3.bucket-name` (and so the new constant `rulerBucketName`). We can store everything is the same bucket as far as `-blocks-storage.storage-prefix` (which is what `CommonStorageBackendFlags()` does).

I would move `ingester.ring.replication-factor: 1` to `New(Read|Write|Backend)Instance()` like we in the microservices `NewXXX()` funcitons, given most tests run with RF=1.

**This is the only comment I would suggest you to address in this PR.**

Reply:
Fixed in 4ffe6a8, thanks for the suggestion.